xGame/Java/xgame/src/main/java/com/xgame/service/UserService.java

```diff
@@ -70,13 +76,24 @@ public void deactivateUser(Integer id) {
 
 	@Override
 	public ProfileViewModel getProfile(Integer id) {
-		// TODO Auto-generated method stub
-		return null;
-	}
-
-	@Override
-	public ProfileViewModel getProfile(String email) {
-		// TODO Auto-generated method stub
-		return null;
+		var u = userRepo.findById(id);
+		
+		if (!u.isPresent()) {
+			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "No account found for given email and password.");
+		}
+		
+		var user = u.get();
+		
+		var matches = matchRepo.findByBlackPlayerIdAndMatchStatus(user.getId(), MatchStatus.COMPLETED);
+		matches.addAll(matchRepo.findByWhitePlayerIdAndMatchStatus(user.getId(), MatchStatus.COMPLETED));
```

Comment:
fyi. you could also make a repo method like:
`matchRepo.FindByWhitePlayerIdOrBlackPlayerIdAndMatchStatus`